File: python/5.PrefixSums/GenomicRangeQuery.py
def solution(S, P, Q):  
    count_a = [0] * (len(S) + 1)
    count_c = [0] * (len(S) + 1)
    count_g = [0] * (len(S) + 1)
    count_t = [0] * (len(S) + 1)

    counter = 1;
    for char in S:
        if char == 'A':
            count_a[counter] += 1;
        if char == 'C':
            count_c[counter] += 1;
        if char == 'G':
            count_g[counter] += 1;
        if char == 'T':
            count_t[counter] += 1;
        if counter > 0:
            count_a[counter] += count_a[counter -1]
            count_c[counter] += count_c[counter -1]
            count_g[counter] += count_g[counter -1]
            count_t[counter] += count_t[counter -1]
        counter+=1
    output = [0] * len(P)
    for i in range(len(P)):
        start = P[i]
        end = Q[i] + 1
        a_diff = count_a[end] - count_a[start]
        c_diff = count_c[end] - count_c[start]
        g_diff = count_g[end] - count_g[start]
        t_diff = count_t[end] - count_t[start]
        if a_diff > 0:
            output[i] = 1
        elif c_diff > 0:
            output[i] = 2
        elif g_diff > 0:
            output[i] = 3
        elif t_diff > 0:
            output[i] = 4
    return output

File: csharp/5.PrefixSums/GenomicRangeQuery.cs
using System;
// you can also use other imports, for example:
// using System.Collections.Generic;

// you can write to stdout for debugging purposes, e.g.
// Console.WriteLine("this is a debug message");

class Solution {
    public int[] solution(string S, int[] P, int[] Q)
    {
        int[] countA = new int[S.Length + 1];
        int[] countC = new int[S.Length + 1];
        int[] countG = new int[S.Length + 1];
        int[] countT = new int[S.Length + 1];

        int counter = 1;
        foreach (char c in S)
        {
            if (c == 'A')
            {
                countA[counter]++;
            }
            if (c == 'C')
            {
                countC[counter]++;
            }
            if (c == 'G')
            {
                countG[counter]++;
            }
            if (c == 'T')
            {
                countT[counter]++;
            }
            if (counter > 0)
            {
                countA[counter] += countA[counter - 1];
                countC[counter] += countC[counter - 1];
                countG[counter] += countG[counter - 1];
                countT[counter] += countT[counter - 1];
            }
            counter++;
        }
        int[] output = new int[P.Length];
        for (int i = 0; i < P.Length; i++)
        {
            int start = P[i];
            int end = Q[i] + 1;
            int aDiff = countA[end] - countA[start];
            int cDiff = countC[end] - countC[start];
            int gDiff = countG[end] - countG[start];
            int tDiff = countT[end] - countT[start];
            if (aDiff > 0)
            {
                output[i] = 1;
            }
            else if (cDiff > 0)
            {
                output[i] = 2;
            }
            else if (gDiff > 0)
            {
                output[i] = 3;
            }
            else if (tDiff > 0)
            {
                output[i] = 4;
            }
        }
        return output;
    }
}

